atc/creds/vault/vault.go

```diff
@@ -41,14 +42,30 @@ func (v Vault) NewSecretLookupPaths(teamName string, pipelineName string, allowR
 
 // Get retrieves the value and expiration of an individual secret
 func (v Vault) Get(secretPath string) (interface{}, *time.Time, bool, error) {
-	secret, expiration, found, err := v.findSecret(secretPath)
+	var secretName string
+	secretRealPath := secretPath
+	if strings.Contains(secretPath, ".") {
```

Comment:
Hmm, I don't think this approach will work. Our `vars/` implementation is already supposed to handle `foo.bar` by splitting out the `foo`, so the `Get` call should only be called with `"foo"`. The extracting of fields off of the value is then applied to the secret that was returned by `Get` - the `secret.Data["value"]` is a special-case.

Reply:
actually for use as an example of secret we have json object with multiple secrets like:
{
git_token:"xxx"
webhook_token: "xxx"
concourse_webhook_pr: "xxx"
}
The problem with shared path which I triying to fix, if you have for example several defaults on different levels, like pipeline/team/root. And concourse is trying to find the key called default.concourse_webhook_pr in the vault, when it will lookup for default for example in the pipeline level it will find the secret, and will return as found, but when it will try to get  concourse_webhook_pr key from this map, it's failing with error  `failed to interpolate task config: Finding variable 'defaults.concourse_webhook_pr': missing field 'concourse_webhook_pr' in var: defaults.concourse_webhook_pr` and will not go through other available path. It still up to you if it suppose to work like this or not, but as for me in vault it's a pretty common case since secret.Data is returning map

Reply:
Yeah, I still don't really like that pattern; to me it's confusing for `((foo.bar))` to sometimes refer to `foo` in different locations. Utilizing fields in Vault is certainly common, but repeating the same definition name at multiple levels but with different sets of fields feels very error-prone and kind of unconventional. Wouldn't it be better to just explicitly name one `global_defaults` or something?

Reply:
the problem for me is like this, if you have a big organisation and you have team/pipeline/global level of defaults, for example git token per team can be in team defaults, and in the same time pipeline defaults namespace can contains pipeline specific defaults. With this level of lookups it's easier as for me to have some standardised pipelines definitions. And in the same time keep you vault name spaces more structured. And in scope of vault it will just try to find target key across all available folders, for me it's pretty much the same what's happening if you secret is looks like `{ value: "xxx"}`